doomed_patient/doomed_patient/NPC.cpp

```diff
@@ -10,18 +10,107 @@ NPC::NPC(PatientGame* game, VectorXY startCoordinates, Texture * sprite)
 	int random = rand() % 4;
 	movementDirection = static_cast<Directions::Direction>(random);
 }
+
+
 void NPC::move(Directions::Direction direction)
 {
 	centre += Directions::getDirectionVector(direction) * speed;
 }
+
+
 void NPC::update(LevelCell::Passable blocked)
 {
+	// Makes sure NPC's grid position is up to date
+	updateCurrentCell();
+
+	// Update direction if necessary
+	updateDirection();
+
+	move(movementDirection);
+}
+
+// Call NPC behaviour code/methods that change NPC movementDirection from this method!
+void NPC::changeDirection()
+{
+	std::shared_ptr<CellEdge> currentEdge = currentCell->getEdge(movementDirection);
+
+
+	//PUT STUFF HERE
+
+
+	// Temporary for testing
+	// Changes direction to a random direction until it finds one that isn't a wall
+	while (currentEdge->isWall())
```

Comment:
This loop just changes the NPCs direction randomly if it hits a wall, only put it in for testing - delete it and then put your stuff in